client/src/components/HistoryViewer/HistoryViewerVersion.js

```diff
@@ -38,22 +38,30 @@ class HistoryViewerVersion extends Component {
    * When clicking on a version, render the detail view for it via a Redux action dispatch
    */
   handleClick() {
-    const { onSelect, version, isActive } = this.props;
+    const { onSelect, version, isActive, compareMode, compareFrom, onCompareSelect } = this.props;
 
     // If the clear button is shown, don't do anything when clicking on the row
     if (isActive) {
       return;
     }
 
-    onSelect(version.Version);
+    if (compareMode) {
+      onCompareSelect(version.Version, !compareFrom);
+    } else {
+      onSelect(version.Version);
```

Comment:
I thought about that briefly, but was unsure if props (and roughly thus state) are available at that point (the map closure not being in the class with props assigned and all that). `dispatch` function is the only thing (currently) being passed into the mapper, if there's a second parameter with the state then this would be possible sure :)